File: tourmatefe/types/customer.ts
import { Account } from "./account";

export type Customer = {
  customerId: number;
  fullName: string;
  accountId: number;
  address: string;
  gender: string;
  dateOfBirth: string;
  phone: string;
  image: string;
  account: Account;
};

export type CustomerAdminUpdateModel = {
    customerId: number,
    fullName: string,
    gender: string,
    dateOfBirth: string,
    accountId: number,
    phone: string,
    email: string,
    password?: string
}

File: TourMateRemoveMembership/Repositories/Models/Customer.cs
using System;
using System.Collections.Generic;

namespace Repositories.Models;

public partial class Customer
{
    public int CustomerId { get; set; }

    public string FullName { get; set; } = null!;

    public int AccountId { get; set; }

    public string Gender { get; set; } = null!;

    public DateOnly DateOfBirth { get; set; }

    public string Phone { get; set; } = null!;

    public string? Image { get; set; }

    public virtual Account Account { get; set; } = null!;

    public virtual ICollection<Feedback> Feedbacks { get; set; } = new List<Feedback>();

    public virtual ICollection<Invoice> Invoices { get; set; } = new List<Invoice>();
}
